From owner-linux-pci@atrey.karlin.mff.cuni.cz Mon Aug 27 16:17:49 2007
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Mon, 27 Aug 2007 16:17:38 -0700
Subject: cpqphp: Use PCI_CLASS_REVISION instead of PCI_REVISION_ID for read
To: gregkh@suse.de
Cc: kristen.c.accardi@intel.com, linux-pci@atrey.karlin.mff.cuni.cz
Message-ID: <20070827231707.27448.22421.stgit@localhost.localdomain>


While PCI_CLASS_REVISION and PCI_REVISION_ID are identical, the
code here discards the revision content and is only interested in
the class number.

Signed-off-by: Auke Kok <auke-jan.h.kok@intel.com>
Cc: Krosten Accardi <kristen.c.accardi@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/hotplug/cpqphp_core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/pci/hotplug/cpqphp_core.c
+++ b/drivers/pci/hotplug/cpqphp_core.c
@@ -549,7 +549,7 @@ get_slot_mapping(struct pci_bus *bus, u8
 			 * slot. */
 			bus->number = tbus;
 			pci_bus_read_config_dword(bus, PCI_DEVFN(tdevice, 0),
-						PCI_REVISION_ID, &work);
+						PCI_CLASS_REVISION, &work);
 
 			if ((work >> 8) == PCI_TO_PCI_BRIDGE_CLASS) {
 				pci_bus_read_config_dword(bus,
